bob/src/hw_metrics_collector.rs

```diff
@@ -133,14 +145,199 @@ impl HWMetricsCollector {
             }
         }
 
-        DisksMetrics {
+        DisksSpaceMetrics {
             total_space: total,
             used_space: used,
             free_space: free
         }
     }
 }
 
+struct CPUStatCollector {
+    iostat_avl: bool,
+}
+
+impl CPUStatCollector {
+    fn new() -> CPUStatCollector {
+        CPUStatCollector { iostat_avl: true }
+    }
+
+    fn parse_iostat_result(iostat_str: &str) -> Result<f64, std::num::ParseFloatError> {
+        iostat_str.replace(',', ".").parse()
+    }
+
+    fn iowait(&mut self) -> Result<f64, String> {
+        if !self.iostat_avl {
+            return Err("iostat is not awailable".to_string());
+        }
+
+        let ios = std::process::Command::new("iostat").arg("-c").output();
+        match ios {
+            Ok(output) => {
+                if let (true, Ok(out)) = (output.status.success(), String::from_utf8(output.stdout))
+                {
+                    // find avg-cpu headers line
+                    let mut lines = out
+                        .lines()
+                        .skip_while(|line| line.find("avg-cpu:").is_none());
+                    // find iowait column on the next line
+                    if let Some(line) = lines.next() {
+                        let mut values = line.split_whitespace();
+                        // iowait is in 3d column
+                        if let Some(iowait_str) = values.nth(3) {
+                            if let Ok(iowait) = Self::parse_iostat_result(iowait_str) {
+                                return Ok(iowait);
+                            }
+                        }
+                    }
+                }
+            }
+            Err(e) => {
+                debug!("Failed to execute iostat: {}", e);
+                self.iostat_avl = false;
+            }
+        };
+        Err("iostat is not awailable".to_string())
+    }
+}
+
+struct DiskStatCollector {
+    iostat_avl: bool,
+    disk_metric_data: Vec<(String, String)>,
+}
+
+impl DiskStatCollector {
+    fn new(disks: &HashMap<PathBuf, String>) -> Self {
+        let mut disk_metric_data = Vec::new();
+        for (path, disk_name) in disks {
+            let path_str = path.as_os_str().to_str().unwrap();
+            if let Ok(dev_name) = Self::dev_name(path_str) {
+                let metric_prefix = format!("{}.{}", DISKS_FOLDER, disk_name);
```

Comment:
This metrics should be in `hardware.disks` group. See https://github.com/qoollo/bob/issues/452